client/src/attestation.rs

```diff
@@ -111,46 +124,49 @@ impl AttestationPayload {
 	}
 }
 
-pub fn recover_ethereum_address(
+/// Recover the signing Ethereum address from a signed attestation
+pub fn address_from_signed_att(
 	signed_attestation: &SignedAttestation,
-) -> Result<ethers::core::types::Address, &'static str> {
+) -> Result<Address, &'static str> {
+	// Get the signing key
 	let public_key = signed_attestation.recover_public_key()?;
-	let public_key_bytes = public_key.serialize_uncompressed();
-	let hashed_public_key = keccak256(&public_key_bytes[1..]);
-	let address_bytes = &hashed_public_key[hashed_public_key.len() - 20..];
-	let address = ethers::core::types::Address::from_slice(address_bytes);
 
-	Ok(address)
+	// Get the address from the public key
+	address_from_public_key(&public_key)
 }
 
-pub fn get_contract_attestation_data(
+/// Construct the contract attestation data from a signed attestation
+pub fn att_data_from_signed_att(
 	signed_attestation: &SignedAttestation,
 ) -> Result<ContractAttestationData, &'static str> {
 	// Recover the Ethereum address from the signed attestation
-	let address = recover_ethereum_address(signed_attestation)?;
+	let address = address_from_signed_att(signed_attestation)?;
 
 	// Calculate the hash of the attestation
 	let attestation_hash = signed_attestation.attestation.hash().to_bytes();
 
-	// Get the signature bytes
-	let signature = signed_attestation.signature.serialize_compact().1;
+	// Get the payload bytes
+	let payload = AttestationPayload::from_signed_attestation(&signed_attestation)?;
 
 	Ok(ContractAttestationData(
 		address,
 		attestation_hash,
```

Comment:
Just noticed this, sorry.

The second parameter in `ContractAttestationData` is `key`, which should be taken from `signed_attestation.attestation`

Reply:
Oh thanks! Good catch, just fixed this.